walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -88,66 +114,90 @@
    * @param duration the duration of the meeting request
    * @param availableTimes the arraylist that we can add a time to
    */
-  public static void addAvailableTime(int start, int prevEnd, int duration,
+  public static boolean addAvailableTime(int start, int prevEnd, int duration,
       ArrayList<TimeRange> availableTimes) {
     if (start > prevEnd) {
       int availableDuration = start - prevEnd;
       if (availableDuration >= duration) {
         TimeRange availableTime = TimeRange.fromStartDuration(prevEnd, 
             availableDuration);
         availableTimes.add(availableTime);
+        return true;
       }
     }
+    return false;
   }
 
   /**
    * Returns the times available to have the meeting of a certain duration
    * Time Complexity: O(n)
    *
-   * @param times Contains an arraylist of an array of start and end times
-   *    this array should be ordered and contain no duplicate start times
+   * @param times Contains an ordered arraylist of an array of start and end 
+   *     times this array should be ordered and contain no duplicate start times
    * @param duration the duration of time the meeting request lasts
    * @return an arraylist of times available whose duration are >= duration
    */
-  public static Collection<TimeRange> timeAvailable(ArrayList<int[]> times, 
+  public ArrayList<TimeRange> timeAvailable(ArrayList<int[]> times, 
       int duration) {
-    int endTime = TimeRange.START_OF_DAY;
+    int startTime = TimeRange.START_OF_DAY;
+    int prevEndTime = TimeRange.START_OF_DAY;
```

Comment:
Add units to these variables. For example, if time is stored in minutes since midnight, specify it: `startTimeMinutes` etc.